### PR TITLE
create a combine LiveData Transformation

### PR DESCRIPTION
**Pull request by frett**
I got tired of using similar logic and having to think through it every time I used it...

example usage:
```kotlin
val data1: LiveData<String>
val data2: LiveData<String>
val combined: LiveData<String> = data1.combineWith(data2) { x, y -> x + y }
```

### CODE REVIEW

File: gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import androidx.lifecycle.Observer
+
+/**
+ * This method will combine 2 LiveData objects into a new LiveData object by running the {@param mapFunction} on the
+ * current values of both source LiveData objects.
+ * Returning either of the original LiveData objects will cause an Exception.
+ */
+@JvmName("combine")
+fun <X, Y, Z> LiveData<X>.combineWith(other: LiveData<Y>, mapFunction: (X?, Y?) -> LiveData<Z>): LiveData<Z> {
+    val result = MediatorLiveData<Z>()
```
Comment (gyasistory) [inline gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt]: Can this be used to combine two LiveData calls?

Comment (frett) [reply gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt]: depends on what you mean by that, this is based off of `Transformations.switchMap()`

Comment (gyasistory) [reply gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt]: Ok, so not what I was thinking of.  I was thinking more on the lines of an `Observable.zip()` replacement.

Comment (frett) [reply gto-support-lifecycle/src/main/java/org/ccci/gto/android/common/lifecycle/Transformations2.kt]: you could utilize it for something like that. I had trouble figuring out a good name for this method. As it stands I'm not really sold on the `combine`/`combineWith` name, but I don't know of anything better.

**Pull request merged**